tests/integration/publish/test_command_integ.py

```diff
@@ -34,65 +37,39 @@ def tearDown(self):
         # Delete application for each test
         self.sar_client.delete_application(ApplicationId=self.application_id)
 
-    def test_update_application(self):
-        template_path = self.temp_dir.joinpath("template_update_app.yaml")
-        command_list = self.get_command_list(template_path=template_path, region=self.region_name)
-
-        process = Popen(command_list, stdout=PIPE)
-        try:
-            stdout, _ = process.communicate(timeout=TIMEOUT)
-        except TimeoutExpired:
-            process.kill()
-            raise
-        process_stdout = stdout.strip()
-
-        expected_msg = 'The following metadata of application "{}" has been updated:'.format(self.application_id)
-        self.assertIn(expected_msg, process_stdout.decode("utf-8"))
-
-        app_metadata_text = self.temp_dir.joinpath("metadata_update_app.json").read_text()
-        app_metadata = json.loads(app_metadata_text)
-        self.assert_metadata_details(app_metadata, process_stdout.decode("utf-8"))
-
-    def test_create_application_version(self):
-        template_path = self.temp_dir.joinpath("template_create_app_version.yaml")
+    @parameterized.expand(
+        [
+            ("template_update_app.yaml", "metadata_update_app.json"),
+            ("template_create_app_version.yaml", "metadata_create_app_version.json"),
+            ("template_create_app_with_readme_body.yaml", "metadata_create_app_with_readme_body.json"),
+        ]
+    )
+    def test_update_application(self, template_filename, expected_template_filename):
+        template_path = self.temp_dir.joinpath(template_filename)
         command_list = self.get_command_list(template_path=template_path, region=self.region_name)
 
-        process = Popen(command_list, stdout=PIPE)
-        try:
-            stdout, _ = process.communicate(timeout=TIMEOUT)
-        except TimeoutExpired:
-            process.kill()
-            raise
-        process_stdout = stdout.strip()
-
+        result = run_command(command_list)
         expected_msg = 'The following metadata of application "{}" has been updated:'.format(self.application_id)
-        self.assertIn(expected_msg, process_stdout.decode("utf-8"))
+        result_msg = result.stdout.decode("utf-8")
+        self.assertIn(expected_msg, result_msg)
 
-        app_metadata_text = self.temp_dir.joinpath("metadata_create_app_version.json").read_text()
+        app_metadata_text = self.temp_dir.joinpath(expected_template_filename).read_text()
         app_metadata = json.loads(app_metadata_text)
-        self.assert_metadata_details(app_metadata, process_stdout.decode("utf-8"))
+        self.assert_metadata_details(app_metadata, result_msg)
 
-    def test_create_application_version_with_semantic_version_option(self):
+    def test_update_application_version_with_semantic_version_option(self):
```

Comment:
This tests creates not updates an application right?

Reply:
I don't think so. It is under the class `TestPublishExistingApp` and the expected message is "The following metadata of application "{}" has been updated." So I corrected the test name to "update" from "create"